lib/perl/Genome/Model/CwlPipeline/Command/Run.pm

```diff
@@ -216,48 +221,103 @@ sub run_cromwell_gcp {
 
-    $self->debug_message('input yaml: %s', $yaml);
-    $self->debug_message('cloud yaml: %s', $cloud_yaml);
-
     # Zip dependencies
-    my $deps_zip = File::Spec->join($data_dir, 'deps.zip');
+    my $deps_zip_path = File::Spec->join($data_dir, 'deps.zip');
+    my $deps_zip_url = "gs://$bucket/build.$build_id/deps.zip";
     my $prev_dir = getcwd;
     my(undef, $deps_dir, undef) = File::Spec->splitpath($main_workflow_file);
+
     chdir($deps_dir);
-    Genome::Sys->shellcmd(cmd => ['zip', '-r', $deps_zip, '.']);
+    Genome::Sys->shellcmd(
+        cmd => ['zip', '-r', $deps_zip_path, '.'],
+        redirect_stdout => '/dev/null'
+        );
     chdir($prev_dir);
 
-    my $workflow_id = Genome::Cromwell->submit_workflow(
-        $main_workflow_file, $cloud_yaml, $deps_zip, $workflow_options )->{id};
+    # Upload zip file
+    Genome::Sys::LSF::bsub::bsub(
+        queue => $queue,
+        user_group => $user_group,
+        resource_string => 'rusage[internet2_download_mbps=500]',
+        wait_for_completion => 1,
+        log_file => File::Spec->join($logdir, '01_upload_zip.log'),
+        cmd => ['gsutil', 'cp', '-n', $deps_zip_path, $deps_zip_url]
+        );
 
-    $self->status_message('Submitted workflow with ID: %s', $workflow_id);
+    #
+    # Generate files for run
+    #
+    my $conf_file = $self->_generate_cromwell_config_gcp($tmp_dir);
+    my $options_file = $self->_generate_workflow_options_gcp;
+    my $labels_file = $self->_generate_cromwell_labels;
 
-    # Poll until done
-    my $status;
+    #
+    # Do the run
+    #
+    $self->status_message("Files generated. Starting VM.");
+    Genome::Sys::LSF::bsub::bsub(
+        queue => $queue,
+        user_group => $user_group,
+        wait_for_completion => 1,
+        log_file => File::Spec->join($logdir, '02_vm_start.log'),
+        cmd => [
+            "sh", "/opt/gms/start.sh",
+            "--build", $build_id,
+            "--cromwell-conf", $conf_file,
+            "--service-account", $cromwell_service_account,
+            "--workflow-definition", $main_workflow_file,
+            "--workflow-inputs", $cloud_yaml,
+            "--workflow-options", $options_file,
+            "--deps-zip", $deps_zip_url,
+            "--bucket", $bucket,
+            "--memory-gb", $cromwell_server_memory_gb,
+            "--tmp-dir", $tmp_dir
+        ] );
+
+    my $result;
+    # Wait for instance VM to terminate itself
+    $self->status_message("VM started. Polling every $poll_interval_seconds seconds.");
     do {
         sleep $poll_interval_seconds;
-        $status = Genome::Cromwell->status($workflow_id)->{status};
-    } while ($status eq "Running" || $status eq "Submitted");
+        $result = system("gcloud compute instances describe build-$build_id --zone us-central1-c > /dev/null");
+        $self->status_message("Polled VM and got result $result");
+    } while ($result == 0);
+    $self->status_message("Polling done. Pulling artifacts.\n");
```

Comment:
The `\n` isn't required here.  (As a side note, be aware that the various `*_message` methods also act like `printf`, so you could do `debug_message('Hello, %s', $name)` if y'like.)